src/Microsoft.AspNet.Mvc.TagHelpers/Internal/AttributeMatcher.cs

```diff
@@ -23,7 +22,7 @@ public static class AttributeMatcher
         /// <returns>The <see cref="ModeMatchResult{TMode}"/>.</returns>
         public static ModeMatchResult<TMode> DetermineMode<TMode>(
             TagHelperContext context,
-            IEnumerable<ModeAttributes<TMode>> modeInfos)
+            ModeAttributes<TMode>[] modeInfos)
```

Comment:
Would something less specific such as `IReadOnlyList<ModelAttributes>` really be that much more expensive? This looks bizarre.